libs/language-server/src/lib/ast/io-type.ts

```diff
@@ -0,0 +1,27 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import { strict as assert } from 'assert';
+
+import { BlocktypeInput, BlocktypeOutput } from './generated/ast';
+
+export enum IOType {
+  NONE = 'None',
+  FILE = 'File',
+  TEXT_FILE = 'TextFile',
+  FILE_SYSTEM = 'FileSystem',
+  SHEET = 'Sheet',
+  TABLE = 'Table',
+}
+
+export function getIOType(blockIO: BlocktypeInput | BlocktypeOutput): IOType {
+  const ioName = blockIO.iotype.ref?.name as string;
+
+  assert(
+    Object.values(IOType).some((type) => type === ioName),
+    `IOType ${ioName} does not exist.`,
+  );
+
+  return ioName as IOType;
+}
```

Comment:
I looked this method up and couldn't find it in the code base on main.
- `ioName` can potentially be undefined I think.
- The conversion to the enum can be a little error prone, but that is just me being very nitpicky here. In case an invalid value survives the assert above (which obviously should not be the case), then the program just continues using the invalid string. With the version below it resolves to `undefined`, making discovering such a bug easier.

```suggestion
export function getIOType(blockIO: BlocktypeInput | BlocktypeOutput): IOType {
  const ioTypeName = blockIO.iotype.ref?.name;
  assert(ioTypeName !== undefined);

  assert(
    Object.values(IOType).some((type) => type === ioTypeName),
    `IOType ${ioTypeName} does not exist.`,
  );

  return IOType[ioTypeName as keyof typeof IOType];
}
```